router/shop.py

```diff
@@ -1,128 +1,106 @@
 import uuid
+from typing import Annotated
 
 from fastapi import APIRouter, Depends, status
 from fastapi.encoders import jsonable_encoder
 from fastapi.responses import JSONResponse
 
-from model.shop import ReturnShop, CreateShopForm, ReturnCreateShop, UpdateShopForm
+from model.account import Account
 from model.general import SuccessModel, ErrorModel
-
-from utils.db_process import get_all_results, execute_query, dict_to_sql_command, dict_delete_none
+from model.shop import (
+    ReturnShop, CreateShopForm, ReturnCreateShop, UpdateShopForm,
+)
+from utils import auth
+from utils.db_process import (
+    get_all_results, execute_query, dict_to_sql_command, dict_delete_none, if_exists_in_db,
+)
 
 router = APIRouter(
     tags=["shop"]
 )
 
+async def prepare_response(result, status_code=status.HTTP_200_OK):
+    data = {}
+    if result:
+        data.update({"data": result})
+        message = "Success"
+    else:
+        message = "Fail"
+
+    content = jsonable_encoder({"msg": message, **data})
+
+    return JSONResponse(status_code=status_code, content=content)
+
```

Comment:
function `create_shop` and `update_shop` , `if (result)` is evaluated inside and outside `prepare_response` which is kinda smelly.
More importantly, in function `get_shop`, `prepare_response` can return JSONResponse(status_code=status.HTTP_200_OK, content=content) when result is `false`